### PR TITLE
Remove a membership port FAQ

### PR DESCRIPTION
**Pull request by yux0**
## What does this PR do?
Remove a membership port FAQ

## Notes to reviewers
The issue is resolved https://github.com/temporalio/temporal/issues/1234. We don't need this special instruction.

### CODE REVIEW

File: docs/server/production-deploy.md [hunk 1]
```diff
@@ -161,16 +161,6 @@ You may sometimes want to have multiple parallel deployments on the same cluster
 - when you want to split Temporal deployments based on namespaces, e.g. staging/dev/uat, or for different teams who need to share common infrastructure.
 - when you need a new deployment to change `numHistoryShards`.
 
-**We recommend not doing this if you can avoid it**. If you need to do it anyway, double-check the following:
```
Comment (yiminc) [inline docs/server/production-deploy.md:164]: We still need this for per-1.19 servre. I think we should keep this doc with note that if you are running v1.19+, you don't need to do that.

Comment (yux0) [reply docs/server/production-deploy.md:164]: Updated

File: docs/server/production-deploy.md [hunk 2]
```diff
@@ -161,6 +161,8 @@ You may sometimes want to have multiple parallel deployments on the same cluster
 - when you want to split Temporal deployments based on namespaces, e.g. staging/dev/uat, or for different teams who need to share common infrastructure.
 - when you need a new deployment to change `numHistoryShards`.
 
+You can skip the following guide if your server is running on release v1.19+. V1.19 makes sure the membership from different clusters will not join together.
```
Comment (djmagee) [inline docs/server/production-deploy.md:164]:
```suggestion
You can skip the following procedure if your server is running v1.19 or later.
The v1.19 release ensures that the membership from different clusters does not combine.
```

Comment (yux0) [reply docs/server/production-deploy.md]: Accepted the commit

**Review by djmagee — APPROVED** [review]

**Pull request merged**